### ISSUE TITLE
Error in user deletion : delete_user_service.rb

### ISSUE THREAD

**Issue by RonanPresle** [gitlabhq/gitlabhq#9394]
As an Admin user I suppressed an account and found a error:

NameError (undefined local variable or method 'current_user' for #<DeleteUserService:0x00006fb7be8a08>):
  app/services/delete_user_service.rb:10:in 'block in execute'
  app/services/delete_user_service.rb:7:in 'execute'
  app/controllers/admin/users_controller.rb:89:in 'destroy'

As it can be seen the error is in  app/services/delete_user_service.rb where `current_user` must be replaced by `user`

**Comment by stanhu** [comment]
Does this still happen? I wasn't able to reproduce the issue; users deleted fine with no errors.

`current_user` is technically the correct parameter to the service, although it's not actually being used.
I'm not quite sure it's not resolving.

What version of GitLab are you using? Installing from omnibus or source?

**Comment by RonanPresle** [comment]
Ok I thought it was just a little error and didn't describe much.
I upgraded my instance from 6.X to 7.12.0.rc1 from source. Everything goes quite find but i got trouble getting it working. Finally, it works but the process doesn't need to be calledd via /etc/init.d/gitlab. It is just starting when a call is made on the right URL. It is strange to me but it works. 

I tried to add the omniauth for Github. I added a user this way and tried to remove it. There it fails giving the previous error.

I tried again creating and suppressing a user. If the user is created by admin or omniauth, I can suppress it right after (if the user is completely empty (no project). But if I create a user by omniauth then **fork** another project that will throw an error.
